pkg/virt-handler/ratelimitcmd/backoff.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package ratelimitcmd
+
+import (
+	"math"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// limitedBackoff provides backoff rate limiter with limit functionality,
+// when the limit is reached it stops to calculate next backoff.
+type limitedBackoff struct {
+	clock       clock.Clock
+	limit       time.Duration
+	backoff     wait.Backoff
+	maxStepTime time.Time
+	stepEnd     time.Time
+}
+
+// Ready return true when the current backoff end-time passed and limit is not reached.
+func (l *limitedBackoff) Ready() bool {
+	now := l.clock.Now()
+	return now.After(l.stepEnd) && now.Before(l.maxStepTime)
+}
+
+// Step calculates the next backoff.
+func (l *limitedBackoff) Step() {
+	if !l.Ready() {
+		return
+	}
+	l.stepEnd = l.clock.Now().Add(l.backoff.Step())
+}
+
+func (l *limitedBackoff) StepEnd() time.Time {
+	return l.stepEnd
+}
+
+const DefaultMaxStep = 5 * time.Hour
+
+func NewFixedExponentialLimitedBackoffWithClock(limit time.Duration, clk clock.Clock) limitedBackoff {
```

Comment:
Why do you call it `Fixed`?

Reply:
I picked this word in order to describe something that is already configured, fixed is a synonymy of predetermined

Reply:
Well, you do not expose anything else, so this has no value IMO.
You could call it `Default` if there was a `Custom` version of the same, but as we do not have one, I do not see the need of any.

Reply:
I removed `Fixed` part